### PR TITLE
User can register

### PR DESCRIPTION
**Pull request by swardbladpaulo**
[API provides functionality to create user account ](https://www.pivotaltracker.com/n/projects/2478349)
### User Story
```
As an API
In order for users to search through quotes
We would like to authenticate the visitor
```
## Changes proposed in this pull request:
* add new specfile for login in, change name on spec for registration, added settings in the devise token auth initializer
* Creating api folder in spec/requests folder,
* Moving request spec files to spec/requests/api folder, Creating a test for user registration
* Renaming the spec file from get_five_random_trump_quotes to get_a_random_trump_quote.
* Modifying Devise route and making Devise to skip omniauth_callbacks.
* Adding devise_token_auth
* Creating a User model
* Creating a factory bot for users
* Creating a model spec for user model
* Adding and configuring rack-cors.

## Packages/Gems added
* rack-cors
* devise token_auth

### CODE REVIEW

File: app/controllers/api/quotes_controller.rb [hunk 1]
```diff
@@ -1,15 +1,14 @@
 class Api::QuotesController < ApplicationController
-  API_URL = 'https://api.tronalddump.io'.freeze
+  API_URL = 'https://api.tronalddump.io'
```
Comment (tochman) [inline app/controllers/api/quotes_controller.rb]: Don’t remove the .freeze here.

File: app/controllers/api/quotes_controller.rb [hunk 2]
```diff
@@ -1,15 +1,14 @@
...
 
   def show
-    if params[:id] == 'random'
+    if params[:id] == "random"
```
Comment (tochman) [inline app/controllers/api/quotes_controller.rb]: In ruby, use single quotes unless you string interpolate

File: app/controllers/application_controller.rb [hunk 3]
```diff
@@ -1,2 +1,3 @@
 class ApplicationController < ActionController::API
+        include DeviseTokenAuth::Concerns::SetUserByToken
```
Comment (tochman) [inline app/controllers/application_controller.rb]: Please format your code

File: app/models/user.rb [hunk 4]
```diff
@@ -0,0 +1,10 @@
+# frozen_string_literal: true
+
+class User < ActiveRecord::Base
+  extend Devise::Models
+  # Include default devise modules. Others available are:
+  # :confirmable, :lockable, :timeoutable and :omniauthable
```
Comment (tochman) [inline app/models/user.rb]: Please remove comments

File: spec/requests/api/registrations_spec.rb [hunk 5]
```diff
@@ -0,0 +1,88 @@
+# frozen_string_literal: true
+
+RSpec.describe "GET /api/auth", type: :request do
```
Comment (tochman) [inline spec/requests/api/registrations_spec.rb:3]: You are testing a POST request

File: app/controllers/api/quotes_controller.rb (hunk 1 above)
Comment (KyunginNa) [reply app/controllers/api/quotes_controller.rb]: Yes, we put it back

File: app/controllers/api/quotes_controller.rb (hunk 2 above)
Comment (KyunginNa) [reply app/controllers/api/quotes_controller.rb]: We changed them to single quotes.

File: app/controllers/application_controller.rb (hunk 3 above)
Comment (KyunginNa) [reply app/controllers/application_controller.rb]: Yes, this is handled now.

File: app/models/user.rb (hunk 4 above)
Comment (KyunginNa) [reply app/models/user.rb]: They are deleted now.

**Pull request merged**